src/components/GoogleMap.js

```diff
@@ -0,0 +1,118 @@
+import { StyleSheet, View, Dimensions, Alert, TouchableOpacity, Text } from "react-native";
+import React, { useState, useEffect, useRef } from "react";
+import MapView from "react-native-maps";
+import * as Location from "expo-location";
+import axios from "axios";
+
+import CustomPin from "./CustomPin";
+
+import mock from "../../test.json";
+
+export default function GoogleMap() {
+  const [location, setLocation] = useState(null);
+  const [isLocationServiceEnable, setIsLocationServiceEnable] = useState(true);
+  const [pinsData, setPinsData] = useState([]);
+  const mapViewCoordinateRef = useRef();
+
+  useEffect(() => {
+    (async () => {
+      const { status } = await Location.requestForegroundPermissionsAsync();
+
+      if (status !== "granted") {
+        const alertMessage = "위치정보 활용 동의가 필요합니다";
+
+        Alert.alert("알림", alertMessage, [
+          { text: "취소" },
+          { text: "확인", onPress: () => setIsLocationServiceEnable(false) },
+        ]);
+      }
+
+      const currentLocation = await Location.getCurrentPositionAsync({});
+      const {
+        coords: { longitude, latitude },
+      } = currentLocation;
+
+      setLocation({
+        latitudeDelta: 0.015,
+        longitudeDelta: 0.0121,
+        longitude,
+        latitude,
+      });
+    })();
+  }, [isLocationServiceEnable]);
+
+  useEffect(() => {
+    if (location !== null) {
+      setPinsData(mock);
+      //지금은 이건데 나중에 서버랑 데이터 베이스랑 연결되면
+      //RequestPinsData(location);
+    }
+  }, [location]);
+
+  function handleGetPinsData() {
+    RequestPinsData(mapViewCoordinateRef.current);
+  }
+
+  function handleMapViewCoordinate(e) {
+    mapViewCoordinateRef.current = e;
+  }
+
+  async function RequestPinsData(nowCoordinate) {
+    const { longitude, latitude } = nowCoordinate;
+
+    try {
+      const pinData = await axios({
+        method: "get",
+        url: "url",
+        params: {
+          coordinate: { longitude, latitude },
+        },
+      });
```

Comment:
저희 통일된 양식으로 작성 부탁드립니다.
```suggestion
      const pinData = await axios.get("url", ~);
```